cartographer/mapping/internal/constraints/constraint_builder_2d_test.cc

```diff
@@ -79,12 +79,12 @@ TEST_F(ConstraintBuilder2DTest, FindsConstraints) {
   for (int i = 0; i < 2; ++i) {
     EXPECT_EQ(constraint_builder_->GetNumFinishedNodes(), expected_nodes);
     for (int j = 0; j < 2; ++j) {
-      constraint_builder_->MaybeAddConstraint(submap_id, &submap, NodeId{},
+      constraint_builder_->MaybeAddConstraint(submap_id, &submap, NodeId{0, 0},
```

Comment:
Maybe just specify defaults in `NodeId` struct directly?

Reply:
In my eyes, a Node/SubmapId should never be default-initialized (outside of tests).
If you don't set it, you shouldn't create an id.

Reply:
Then maybe delete the default constructor?